airbyte-ci/connectors/pipelines/pipelines/cli/auto_update.py

```diff
@@ -0,0 +1,122 @@
+# HELPERS
+
+import importlib
+import logging
+import os
+import sys
+
+import asyncclick as click
+import requests
+from pipelines import main_logger
+from pipelines.consts import LOCAL_PIPELINE_PACKAGE_PATH
+
+__installed_version__ = importlib.metadata.version("pipelines")
+
+BINARY_UPGRADE_COMMAND = "make tools.airbyte-ci.install"
+DEV_UPGRADE_COMMAND = "make tools.airbyte-ci-dev.install"
+
+
+def _get_os_name():
+    """
+    Returns 'ubuntu' if the system is Linux or 'macos' if the system is macOS.
+    """
+    OS = os.uname().sysname
+    if OS == "Linux":
+        return "ubuntu"
+    elif OS == "Darwin":
+        return "macos"
+    else:
+        # Default to macos as this is just a check, if they are not supported they will find
+        # out at install time.
+        return "macos"
+
+
+def _is_upgrade_available(version: str, is_dev: bool) -> bool:
```

Comment:
```suggestion
def _is_version_available(version: str, is_dev: bool) -> bool:
```
This function logic does not check if an upgrade is actually available, it just checks if the binary for a specific version exists